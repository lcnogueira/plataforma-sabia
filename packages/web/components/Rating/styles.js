import styled from 'styled-components';
import { AiFillStar } from 'react-icons/ai';

export const FillStar = styled(AiFillStar).attrs(({ size, theme: { colors, sizes } }) => ({
	color: colors.primary,
	size: `${size || sizes.mediumIcon}rem`,
}))``;

export const OutlineStar = styled(AiFillStar).attrs(({ size, theme: { colors, sizes } }) => ({
	color: colors.lightGray4,
	size: `${size || sizes.mediumIcon}rem`,
}))``;
