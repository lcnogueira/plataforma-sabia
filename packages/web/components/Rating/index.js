import React from 'react';
import PropTypes from 'prop-types';
import ReactRating from 'react-rating';

import { FillStar, OutlineStar } from './styles';

/**
 * A smart star rating component.
 *
 * @see https://github.com/dreyescat/react-rating
 *
 * @returns {React.Element}
 */
const Rating = ({
	start,
	stop,
	step,
	fractions,
	value,
	readonly,
	onChange,
	onHover,
	onClick,
	size,
}) => (
	<ReactRating
		start={start}
		stop={stop}
		step={step}
		fractions={fractions}
		initialRating={value}
		readonly={readonly}
		onChange={onChange}
		onHover={onHover}
		onClick={onClick}
		emptySymbol={<OutlineStar size={size} />}
		fullSymbol={<FillStar size={size} />}
	/>
);

Rating.propTypes = {
	start: PropTypes.number,
	stop: PropTypes.number,
	step: PropTypes.number,
	fractions: PropTypes.number,
	value: PropTypes.number,
	size: PropTypes.number,
	readonly: PropTypes.bool,
	onChange: PropTypes.func,
	onHover: PropTypes.func,
	onClick: PropTypes.func,
};

Rating.defaultProps = {
	start: 0,
	stop: 5,
	step: 1,
	fractions: 1,
	value: 0,
	size: null,
	readonly: false,
	onChange: () => null,
	onHover: () => null,
	onClick: () => null,
};

export default Rating;
