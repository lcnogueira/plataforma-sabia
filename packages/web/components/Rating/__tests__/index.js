import React from 'react';
import { render } from 'test-utils';
import Rating from '..';

describe('Rating component', () => {
	it('render correctly', () => {
		const { container } = render(<Rating />);

		expect(container).toMatchSnapshot();
	});

	it('render with a custom size', () => {
		const { container } = render(<Rating size={2} />);

		expect(container).toMatchSnapshot();
	});
});
