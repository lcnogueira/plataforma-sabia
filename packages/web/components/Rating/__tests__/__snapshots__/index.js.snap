// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating component render correctly 1`] = `
<div>
  <span
    style="display: inline-block; direction: ltr;"
  >
    <span
      style="cursor: pointer; display: inline-block; position: relative;"
    >
      <span>
        <svg
          class="styles__OutlineStar-sc-118hsvd-1"
          color="hsl(0, 0%, 93%)"
          fill="currentColor"
          height="2rem"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(237, 237, 237);"
          viewBox="0 0 1024 1024"
          width="2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
          />
        </svg>
      </span>
      <span
        style="display: inline-block; position: absolute; overflow: hidden; top: 0px; left: 0px; width: 0%;"
      >
        <svg
          class="styles__FillStar-sc-118hsvd-0"
          color="hsla(29, 94%, 62%)"
          fill="currentColor"
          height="2rem"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(67, 67, 67);"
          viewBox="0 0 1024 1024"
          width="2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
          />
        </svg>
      </span>
    </span>
    <span
      style="cursor: pointer; display: inline-block; position: relative;"
    >
      <span>
        <svg
          class="styles__OutlineStar-sc-118hsvd-1"
          color="hsl(0, 0%, 93%)"
          fill="currentColor"
          height="2rem"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(237, 237, 237);"
          viewBox="0 0 1024 1024"
          width="2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
          />
        </svg>
      </span>
      <span
        style="display: inline-block; position: absolute; overflow: hidden; top: 0px; left: 0px; width: 0%;"
      >
        <svg
          class="styles__FillStar-sc-118hsvd-0"
          color="hsla(29, 94%, 62%)"
          fill="currentColor"
          height="2rem"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(67, 67, 67);"
          viewBox="0 0 1024 1024"
          width="2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
          />
        </svg>
      </span>
    </span>
    <span
      style="cursor: pointer; display: inline-block; position: relative;"
    >
      <span>
        <svg
          class="styles__OutlineStar-sc-118hsvd-1"
          color="hsl(0, 0%, 93%)"
          fill="currentColor"
          height="2rem"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(237, 237, 237);"
          viewBox="0 0 1024 1024"
          width="2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
          />
        </svg>
      </span>
      <span
        style="display: inline-block; position: absolute; overflow: hidden; top: 0px; left: 0px; width: 0%;"
      >
        <svg
          class="styles__FillStar-sc-118hsvd-0"
          color="hsla(29, 94%, 62%)"
          fill="currentColor"
          height="2rem"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(67, 67, 67);"
          viewBox="0 0 1024 1024"
          width="2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
          />
        </svg>
      </span>
    </span>
    <span
      style="cursor: pointer; display: inline-block; position: relative;"
    >
      <span>
        <svg
          class="styles__OutlineStar-sc-118hsvd-1"
          color="hsl(0, 0%, 93%)"
          fill="currentColor"
          height="2rem"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(237, 237, 237);"
          viewBox="0 0 1024 1024"
          width="2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
          />
        </svg>
      </span>
      <span
        style="display: inline-block; position: absolute; overflow: hidden; top: 0px; left: 0px; width: 0%;"
      >
        <svg
          class="styles__FillStar-sc-118hsvd-0"
          color="hsla(29, 94%, 62%)"
          fill="currentColor"
          height="2rem"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(67, 67, 67);"
          viewBox="0 0 1024 1024"
          width="2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
          />
        </svg>
      </span>
    </span>
    <span
      style="cursor: pointer; display: inline-block; position: relative;"
    >
      <span>
        <svg
          class="styles__OutlineStar-sc-118hsvd-1"
          color="hsl(0, 0%, 93%)"
          fill="currentColor"
          height="2rem"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(237, 237, 237);"
          viewBox="0 0 1024 1024"
          width="2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
          />
        </svg>
      </span>
      <span
        style="display: inline-block; position: absolute; overflow: hidden; top: 0px; left: 0px; width: 0%;"
      >
        <svg
          class="styles__FillStar-sc-118hsvd-0"
          color="hsla(29, 94%, 62%)"
          fill="currentColor"
          height="2rem"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(67, 67, 67);"
          viewBox="0 0 1024 1024"
          width="2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
          />
        </svg>
      </span>
    </span>
  </span>
</div>
`;

exports[`Rating component render with a custom size 1`] = `
<div>
  <span
    style="display: inline-block; direction: ltr;"
  >
    <span
      style="cursor: pointer; display: inline-block; position: relative;"
    >
      <span>
        <svg
          class="styles__OutlineStar-sc-118hsvd-1"
          color="hsl(0, 0%, 93%)"
          fill="currentColor"
          height="2rem"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(237, 237, 237);"
          viewBox="0 0 1024 1024"
          width="2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
          />
        </svg>
      </span>
      <span
        style="display: inline-block; position: absolute; overflow: hidden; top: 0px; left: 0px; width: 0%;"
      >
        <svg
          class="styles__FillStar-sc-118hsvd-0"
          color="hsla(29, 94%, 62%)"
          fill="currentColor"
          height="2rem"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(67, 67, 67);"
          viewBox="0 0 1024 1024"
          width="2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
          />
        </svg>
      </span>
    </span>
    <span
      style="cursor: pointer; display: inline-block; position: relative;"
    >
      <span>
        <svg
          class="styles__OutlineStar-sc-118hsvd-1"
          color="hsl(0, 0%, 93%)"
          fill="currentColor"
          height="2rem"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(237, 237, 237);"
          viewBox="0 0 1024 1024"
          width="2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
          />
        </svg>
      </span>
      <span
        style="display: inline-block; position: absolute; overflow: hidden; top: 0px; left: 0px; width: 0%;"
      >
        <svg
          class="styles__FillStar-sc-118hsvd-0"
          color="hsla(29, 94%, 62%)"
          fill="currentColor"
          height="2rem"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(67, 67, 67);"
          viewBox="0 0 1024 1024"
          width="2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
          />
        </svg>
      </span>
    </span>
    <span
      style="cursor: pointer; display: inline-block; position: relative;"
    >
      <span>
        <svg
          class="styles__OutlineStar-sc-118hsvd-1"
          color="hsl(0, 0%, 93%)"
          fill="currentColor"
          height="2rem"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(237, 237, 237);"
          viewBox="0 0 1024 1024"
          width="2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
          />
        </svg>
      </span>
      <span
        style="display: inline-block; position: absolute; overflow: hidden; top: 0px; left: 0px; width: 0%;"
      >
        <svg
          class="styles__FillStar-sc-118hsvd-0"
          color="hsla(29, 94%, 62%)"
          fill="currentColor"
          height="2rem"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(67, 67, 67);"
          viewBox="0 0 1024 1024"
          width="2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
          />
        </svg>
      </span>
    </span>
    <span
      style="cursor: pointer; display: inline-block; position: relative;"
    >
      <span>
        <svg
          class="styles__OutlineStar-sc-118hsvd-1"
          color="hsl(0, 0%, 93%)"
          fill="currentColor"
          height="2rem"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(237, 237, 237);"
          viewBox="0 0 1024 1024"
          width="2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
          />
        </svg>
      </span>
      <span
        style="display: inline-block; position: absolute; overflow: hidden; top: 0px; left: 0px; width: 0%;"
      >
        <svg
          class="styles__FillStar-sc-118hsvd-0"
          color="hsla(29, 94%, 62%)"
          fill="currentColor"
          height="2rem"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(67, 67, 67);"
          viewBox="0 0 1024 1024"
          width="2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
          />
        </svg>
      </span>
    </span>
    <span
      style="cursor: pointer; display: inline-block; position: relative;"
    >
      <span>
        <svg
          class="styles__OutlineStar-sc-118hsvd-1"
          color="hsl(0, 0%, 93%)"
          fill="currentColor"
          height="2rem"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(237, 237, 237);"
          viewBox="0 0 1024 1024"
          width="2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
          />
        </svg>
      </span>
      <span
        style="display: inline-block; position: absolute; overflow: hidden; top: 0px; left: 0px; width: 0%;"
      >
        <svg
          class="styles__FillStar-sc-118hsvd-0"
          color="hsla(29, 94%, 62%)"
          fill="currentColor"
          height="2rem"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(67, 67, 67);"
          viewBox="0 0 1024 1024"
          width="2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
          />
        </svg>
      </span>
    </span>
  </span>
</div>
`;
